RELEASING.md

```diff
@@ -0,0 +1,60 @@
+1. `git checkout -b "release-$FEDERATION_RELEASE_VERSION" main`
+1. Update `CHANGELOG.md` in each repo ([more info](#changelogs))
+1. `git add ':/*CHANGELOG.md'` to add only changelogs to the commit
+1. `git commit --message "Update CHANGELOGs for v$FEDERATION_RELEASE_VERSION release"`
+1. `git push -u origin "release-$FEDERATION_RELEASE_VERSION"`
+1. `open "https://github.com/apollographql/federation/compare/main...release-$FEDERATION_RELEASE_VERSION?quick_pull=1&title=Release+$FEDERATION_RELEASE_VERSION&template=APOLLO_RELEASE_TEMPLATE.md"` to open PR in Github
+
+#### Bump version and tag release
+
+1. `npx lerna version --no-push --force-publish=\* "$FEDERATION_RELEASE_VERSION"` ([more info](#lerna-version))
+    1. `git show --pretty="" --name-only HEAD` to see the files in the commit
+    1. `git --no-pager tag --points-at HEAD` to see the tags that were created
+1. `git push --follow-tags origin` to push the version bumps & tags created by lerna in the previous step
+1. `APOLLO_DIST_TAG=next npm run release:start-ci-publish` ([more info](#publishing))
+1. `open https://app.circleci.com/pipelines/github/apollographql/federation?filter=mine` to approve publishing to NPM
+    1. There will also be a message posted to #team-atlas in slack that has a link to the approval job
+1. `npm view @apollo/subgraph versions` to ensure the version that was just published shows up in the NPM registry
+
+#### Merge the release PR
+
+1. Merge the release PR to `main`
+
+![celebrate](https://media.giphy.com/media/LZElUsjl1Bu6c/giphy.gif)
+
+---
+
+## Changelogs
+
+Changelogs should be updated when creating a release. If there are no changes for a given package, add an entry that states `Released in sync with other federation packages but no changes to this package.`
+
+1. [gateway-js/CHANGELOG.md](gateway-js/CHANGELOG.md) all changes should be reflected in here, since it's essentially a rollup of all packages
+1. [composition-js/CHANGELOG.md](composition-js/CHANGELOG.md)
+1. [subgraph-js/CHANGELOG.md](subgraph-js/CHANGELOG.md)
+1. [query-planner-js/CHANGELOG.md](query-planner-js/CHANGELOG.md)
+1. [internals-js/CHANGELOG.md](internals-js/CHANGELOG.md)
+1. [query-graphs-js/CHANGELOG.md](query-graphs-js/CHANGELOG.md)
+
+## `lerna version`
+
+Learn more about the [`lerna version` command](https://github.com/lerna/lerna/tree/main/commands/version). For our purposes, the important part is that `lerna` will create a tag for each package (e.g. `@apollo/federation-internals@2.0.0-preview.7`).
+
+#### Why does `package-lock.json` get updated?
+
+`npm version` is getting called due to [lerna version lifecycle hooks](https://github.com/lerna/lerna/tree/main/commands/version#lifecycle-scripts), which calls `npm i --package-lock-only` causing the package lock to get regenerated (this is a good thing).
+
+## Publishing
+
+The important part is creating a `publish/*` tag, which starts a job in CircleCI to start the publishing process.
```

Comment:
It's a little confusing to read this, maybe worth mentioning this is handled by the `release:start-ci-publish` script and not something that needs to be done manually.

Reply:
Good call. I should make this a footnote or add more context here.

Reply:
Made this into a footnote, lemme know if that makes it better